cmake/compiler/gcc/target.cmake

```
@@ -6,10 +6,33 @@ set_ifndef(C++ g++)
 # GCC-based toolchains
 
 if("${SPARSE}" STREQUAL "y")
-  find_program(CMAKE_C_COMPILER cgcc)
-  find_program(REAL_CC ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
-  set(ENV{REAL_CC} ${REAL_CC})
-else()
+  # No search PATHS because we need more than cgcc in the default PATH
+  find_program(CMAKE_C_COMPILER cgcc REQUIRED)
+  message(STATUS "sparse configuration: cgcc=${CMAKE_C_COMPILER}")
```

Generally we should not just print variables, it will be a fast track to noise from CMake invocation.

Users can easily see this value directly in the CMake cache.

If you want to print some, then make it a message that conforms to how CMake prints information on found packages, which uses this format:
```
-- Found <package>: <path> (found version "<version>") 
```

as sparse doesn't seem to have version (or at least we don't care for it), then the following will be ok.

```suggestion
  message(STATUS "Found sparse: ${CMAKE_C_COMPILER}")
```

---

Thanks for letting me keep this message.

I updated and kept this `message()` (and only this one) because it's:
- appearing only when -DSPARSE=y = zero added noise for almost all users.
- conveniently signaling a sparse build at the start of the logs
- just before any REAL_CC configuration error

The location is especially useful considering this feature currently requires a not yet released version of sparse.


> Users can easily see this value directly in the CMake cache.

There are a few valid and "less valid" issues with that approach
- Many users actively refuse to learn anything about the build system (https://mesonbuild.com/Design-rationale.html) and have absolutely no idea that a "CMake cache" even exists.
- Searching for "sparse" in the CMake Cache will not find `CMAKE_C_COMPILER`
- Everyone has access to CI logs, no one has access to the CI's CMake cache.

Don't get me wrong: I'm not saying the CMake cache should be entirely printed, very far from it. But I think some overlap may be useful or even necessary sometimes.

---

> Thanks for letting me keep this message.

Information on what is happening in the system is still valuable.
Especially when looking up tools. This is one reason why a `find_package()` in module module mode has a small advantage over a pure `find_program()`.

> Don't get me wrong: I'm not saying the CMake cache should be entirely printed, very far from it.

Unfortunately this is where some project ends.
Cause every person has a different interpretation as to what is important.

As soon as just a single person have experienced problem A, and a message printing state of A would have saved that person, then that value is important enough to print.

> But I think some overlap may be useful or even necessary sometimes.

If we want to print CMake variables in general, then that should be done as:
```
message(DEBUG "<VAR_NAME>: ${VAR_NAME}")
```

using `DEBUG` will allow users who want to know everything to use `cmake --log-level=debug`.
And everyone else will still have a clean log.

> Everyone has access to CI logs, no one has access to the CI's CMake cache.

Feel free to go through all CMake variables that should be of special interest to users and print them using `DEBUG`.
CI can easily execute with `--log-level=debug`.

I'm not against such an approach, but then we should consider to generally do this, and not just for var A and B.
That is, to go through the build system and at least try to find the 25% most important vars, as to get a pattern established.